onnxruntime/core/util/thread_utils.cc

```diff
@@ -144,4 +199,15 @@ ORT_API_STATUS_IMPL(SetGlobalCustomJoinThreadFn, _Inout_ OrtThreadingOptions* tp
   return nullptr;
 }
 
+ORT_API_STATUS_IMPL(SetGlobalIntraOpThreadAffinity, _Inout_ OrtThreadingOptions* tp_options, const char* affinity_string) {
+  if (!tp_options) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "Received null OrtThreadingOptions");
+  }
+  if (!affinity_string || strnlen(affinity_string, onnxruntime::kMaxStrLen) == 0) {
```

Comment:
Should we also check for length > kMaxStrLen?

Reply:
Added the upper bound.